### PR TITLE
Genres as one route

### PR DESCRIPTION
**Pull request by AmerMekic**
# Description

Deleted routes GenreComedy, GenreAction, etc. , made only one Genre route that filters movies genres from url parameter

Task # (task link)

https://app.clickup.com/t/26etff1

## Screenshots

Screenshot 2022-02-21 024625

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Pages/Genre.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import {useState} from 'react';
+import {useParams} from 'react-router-dom';
+import mockData from '../mockdata.json';
+
+export const Genre = () => {
+  let param = useParams().genre;
```
Comment (zlayabekrija) [inline src/Pages/Genre.js]: Please use object destructure on use params hook ```const {genre} = useParams()```

File: src/Pages/Genre.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  param = param[0].toUpperCase() + param.slice(1);
+  let selectedGenre = [];
+  //mockData will be real store data TODO
+  mockData.movies.map((movie) => {
```
Comment (zlayabekrija) [inline src/Pages/Genre.js]: Let's use state for now `const [movies,setMovies] = useState()
useEffect(()=>{
   setMovies(mockData.filter(movie=>movie.genres.includes(genre)))
},[genre])
`

File: src/Pages/Genre.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    if (movie.genres.includes(param)) {
+      selectedGenre.push(movie);
+    }
+  });
+  return (
+    <div>
+      {/*Instead of <p></p> TODO MovieCard*/}
+      {selectedGenre.map((movie) => (
```
Comment (zlayabekrija) [inline src/Pages/Genre.js]: Let's take this from the state 
``movies.map(movie=><p>{movie?.title</p>})

**Review by zlayabekrija — CHANGES_REQUESTED** [review]

File: src/Pages/Genre.js (hunk 3 above)
Comment (AmerMekic) [reply src/Pages/Genre.js]: I will fix all of these per given instructions

File: src/Pages/Genre.js (hunk 1 above)
Comment (AmerMekic) [reply src/Pages/Genre.js]: Done

File: src/Pages/Genre.js (hunk 2 above)
Comment (AmerMekic) [reply src/Pages/Genre.js]: Done

File: src/Pages/Genre.js (hunk 3 above)
Comment (AmerMekic) [reply src/Pages/Genre.js]: Done

**Review by zlayabekrija — APPROVED** [review]

**Review by majraoruc — APPROVED** [review]

**Pull request merged**